storage/innobase/include/row0ftsort.h

```diff
@@ -96,6 +96,7 @@ struct fts_psort_t {
 /** Row fts token for plugin parser */
 struct row_fts_token_t {
 	fts_string_t*	text;		/*!< token */
+	ulint 		position;
```

Comment:
What is the unit of this? Bytes? Characters? Is the first position 0? That should be part of the comment, which is missing here.

Do we really need `ulint` a.k.a. `size_t` here? Would `uint32_t` suffice?

Reply:
Elsewhere in the patch, we are adding some `int position`. That would differ in signedness and typically also in the size on 64-bit systems.

Reply:
> Where does this code come from? MySQL 5.7?

This code indeed comes from MySQL.
I've taken field sizes as they were there, without any additional thinking. But know as I look again, they are indeed inconsistent. It's probably a good idea to use `size_t` for all data sizes. But statement `mach_write_to_4(position, pos)` in `row_merge_fts_doc_tokenize()` assumes data are 4 byte long, while `pos` is declared as `ulint`. So I'm a bit confused. Is limiting to 32 bits is fine?

> partitioning

I'll try to dig deeper into that.

Reply:
`mach_write_to_4()` is best matched with `uint32_t`. As far as I understand, that data will ultimately be written to a data file. Still, it would be good if you did some research and also clarified whether that position is counted in bytes or characters. As you should know, those may be different in UTF-8 and will be different in UCS2 or UTF-16.

Reply:
> and also clarified whether that position is counted in bytes or characters

They are in bytes. I've added a comment for the `position` field and changed its type to `unsigned int`.

Patches currently have some compilation issues with Windows-related CI workers that I'm trying to figure out, so I forgot to mention that. Sorry.